dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -755,7 +759,17 @@ static void AddResponseMessage(ChatCompletionsOptions chatOptions, ChatHistory c
             // Add the original assistant message to the chatOptions; this is required for the service
             // to understand the tool call responses.
             chatOptions.Messages.Add(GetRequestMessage(streamedRole ?? default, content, streamedName, toolCalls));
-            chat.Add(new OpenAIChatMessageContent(streamedRole ?? default, content, this.DeploymentOrModelName, toolCalls, metadata) { AuthorName = streamedName });
+            // Add the result message to the caller's chat history
+            var newChatMessageContent = new OpenAIChatMessageContent(streamedRole ?? default, content, this.DeploymentOrModelName, toolCalls, metadata)
+            {
+                AuthorName = streamedName
+            };
+            // Add the tool call messages to the new chat message content for diagnostics purposes.
```

Comment:
Why is this necessary when toolCalls is already being provided to the content ctor?

Reply:
It's because the message is of a derived type of the `ChatMessageContent` type, which doesn't not have the tool calls property.

Reply:
Isn't this going to interfere with non-diagnostic uses, though? This is the user's ChatHistory object, and this is now changing the ChatMessageContent to now have Items that it didn't previously have. That doesn't seem right. @SergeyMenshykh?

Reply:
> Isn't this going to interfere with non-diagnostic uses, though? This is the user's ChatHistory object, and this is now changing the ChatMessageContent to now have Items that it didn't previously have. That doesn't seem right. @SergeyMenshykh?

@TaoChenOSU, do you use the FunctionCallContent items from the chat message content items for model diagnostics? Or are you adding them because similar functionality exists for non-streaming APIs?  
   
Either way, this change will be required to support function call content classes for streaming. I would have added it anyway, within the scope of the task - https://github.com/microsoft/semantic-kernel/issues/5818, to communicate the fact of the function call by LLM to callers in a connector-agnostic manner.  
   
I don't expect callers to be impacted by the change. On the contrary, they will be able to get function call info from the chat message item's Items collection in a connector-agnostic way rather than casting the chat message to the OpenAI one and accessing the info via the ToolCalls property.

Reply:
It’s required for model diagnostics.